src/Umbraco.Core/Runtime/CoreRuntime.cs

```diff
@@ -225,6 +231,62 @@ protected virtual IFactory Boot(IRegister register, DisposableTimer timer)
             return _factory;
         }
 
+        private void DoUnattendedInstall(IUmbracoDatabaseFactory databaseFactory)
+        {
+            // unattended install is not enabled
+            if (RuntimeOptions.InstallUnattended == false) return;
+
+            var localVersion = UmbracoVersion.LocalVersion; // the local, files, version
+            var codeVersion = _state.SemanticVersion; // the executing code version
+
+            // local version and code version is not equal, an unattended install cannot be performed
+            if (localVersion != codeVersion) return;
+
+            // no connection string set
+            if (databaseFactory.Configured == false) return;
+
+            var tries = 5;
+            var connect = false;
+            for (var i = 0;;)
+            {
+                connect = databaseFactory.CanConnect;
+                if (connect || ++i == tries) break;
+                Logger.Debug<CoreRuntime>("Could not immediately connect to database, trying again.");
+                Thread.Sleep(1000);
+            }
+
+            // could not connect to the database
+            if (connect == false) return;
+
+            using (var database = databaseFactory.CreateDatabase())
+            {
+                var isDatabaseEmpty = databaseFactory.SqlContext.SqlSyntax.GetTablesInSchema(database).Any() == false;
+
+                // database is not empty, assume Umbraco is already installed
+                if (isDatabaseEmpty == false) return;
+
+                // all conditions fulfilled, do the install
+                Logger.Info<CoreRuntime>("Starting unattended install.");
+                database.BeginTransaction();
```

Comment:
This needs to be within the try/catch, creating a trans will throw if for some reason the db connection is severed. I've fixed this up.